### PR TITLE
chore(gitignore): add vscode and intellij ide folders to gitignore

### PR DESCRIPTION
**Pull request by bertyhell**
Ignores the folders that are created when using Visual Studio Code or IntelliJ or Webstorm IDE's

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,8 @@
 .DS_Store
 
+.idea/
+.vscode/
```
Comment (manudeli) [inline .gitignore:4]: Could you explain why we need to remove these directory?

Comment (raon0211) [reply .gitignore:4]: I remember that these directories are sometimes auto-generated by the IDE, which can lead to unintended changes in the repository. While we need to block them, there are also specific settings that we include deliberately. I'll review what our `.gitignore` file needs to include shortly.

Comment (sean-esper) [reply .gitignore:4]: Yeah if you dont ignore them, they will just keep showing up in random prs when people `git add .`

Comment (bertyhell) [reply .gitignore:4]: exactly
Each developer can keep their IDE settings how they like, they don't need to be checked into the repository in my opinion

Comment (raon0211) [inline .gitignore:4]:
```suggestion
```

It appears that IntelliJ makes frequent modifications to the `.idea` directory, whereas the `.vscode` directory sees fewer changes. Furthermore, our `.vscode` directory includes default configurations to accommodate Yarn's PnP (Plug 'n' Play) install strategy. Therefore, I suggest that we only ignore the `.idea` directory.

**Review by raon0211 — APPROVED** [review]

**Comment by raon0211** [comment]
Thanks for your pull request!

**Pull request merged**